### DISCUSSION TITLE
Problem launching app with using WixShellExec

### DISCUSSION THREAD

**Discussion by ghost** [wixtoolset/issues#7915] [Questions]
### WiX Version

4.0.3

### .NET or MSBuild or Visual Studio Version

.Net 8.0.100

### HeatWave Version

1.0.2.1

### Windows Version

Win10 22H2

### Repro Steps

I'm trying to launch my app, after the installation is completed. For this, I use the follwing code in my .wxs file:

    <Property Id="WixShellExecTarget" Value="[#MyApp.exe]" />
    <CustomAction Id="LaunchApplication" DllEntry="WixShellExec" Impersonate="yes" BinaryRef="Wix4UtilCA_X64" />

... this does not work. In the msiexec log, I found this:

    Action 17:20:32: ExitDialog. 
    Action start 17:20:32: ExitDialog.
    Action 17:20:32: ExitDialog. Dialog created
    Action 17:20:35: LaunchApplication. 
    Action start 17:20:35: LaunchApplication.
    Action ended 17:20:35: LaunchApplication. Return value 3.
    DEBUG: Error 2896:  Executing action LaunchApplication failed.

But if I use the following code, it works fine:

    <Property Id="WixShellExecTarget" Value="[INSTALLLOCATION]MyApp.exe" />

**Please take a look at the `Value` attribute.**

... but now, I get a `WIX1077` warning:

> The 'WixShellExecTarget' Property contains '[INSTALLLOCATION]' in its value which is an illegal reference to another property. ...

### Actual Result

I'm not sure, if `[#MyApp.exe]` is correct, and there is a bug somewhere, of if `[INSTALLLOCATION]MyApp.exe` is correct, and the warning is wrong?

**Comment by barnson** [comment]
WiX doesn't use issues for support. We moved this issue to the [GitHub discussions area][gh]. You might also consider asking your question on [Stack Overflow][so].

[gh]: https://github.com/wixtoolset/issues/discussions?discussions_q=category%3AQuestions
[so]: https://stackoverflow.com/questions/tagged/wix

**Comment by robmen** [comment]
Try using `SetProperty` to set the Property instead of initializing it with an embedded Property in the value. I'm a little surprised the nested property works with `WixShellExecTarget` but your log file doesn't look verbose so I don't know that it actually is working.
